src/daily-quest/daily-quest.routes.ts

```diff
@@ -1,20 +1,48 @@
-import { Router } from "express";
+import { RequestHandler, Router } from "express";
 import DailyQuestController from "./daily-quest.controller";
 import { queryParamsParser } from "@/middleware/queryParams";
+import { auth } from "@/middleware/authentication";
+import { routeParamsParser } from "@/middleware/routeParams";
 
 const dailyQuestRouter = Router();
 const dailyQuestController = new DailyQuestController();
 
-dailyQuestRouter.get("/", queryParamsParser, dailyQuestController.getAll);
-dailyQuestRouter.get("/:id", dailyQuestController.getById);
-dailyQuestRouter.post("/", dailyQuestController.create);
+dailyQuestRouter.get(
+  "/",
+  auth,
+  queryParamsParser,
```

Comment:
Add this middleware to route if you want to get query params (sortBy, sortOrder, limit, date, etc) for it.